juju/juju
# Model config client facades

With this change we are now making the model config client facade use DQlite backed services for reading, writing and removing model config values.

All of the model config checking logic bar one item has been moved down into the services layer. The remaining one requires permission checks so we have left that as an API concern.

The storage and secret backend checkers need to be fully implemented as well. This will be worked on with Wallyworld in a follow up PR.

## Checklist


## QA steps

### Test 1
The first test is we need to make sure that model config is the same between this change and the last commit.

1. Bootstrap a controller off of main and add a new model.
2. Run `juju model-config` and save the output.
3. Bootstrap a controller with this branch and add a new model.
4. Run `juju model-config`
5. Compare the output. They should be the same with the exception of storage. This will be dealt with in a follow up PR.

### Test 2
1. Bootstrap a controller and a new model.
2. Set model config values for the new model such as `no-proxy` and confirm they are come back out of the facade with `juju model-config`
3. Test unsetting the model config changes and check the facade reports the changes again.

All other logic tests are satisfied in unit tests.

## Documentation changes

## Links

**Jira card:** JUJU-5342
